spec/order_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'order'
+
+describe Order do
+  let(:menu) {
+    double(
+      :menu,
+      name: "Dominos",
+      dishes:
+      {
+        "Cheese & Tomato" => 6,
+        "Mighty Meaty" => 9,
+        "Pepperoni Passion" => 8,
+        "Tandoori Hot" => 9,
+        "Vegi Sizzler" => 8
+      }
+    )
+  }
+  let(:messenger) { double (:messenger) }
+  subject(:order) { described_class.new(menu, messenger) }
+
+  describe 'initialization' do
+    it 'has an empty basket' do
+      expect(order.basket).to be_empty
+    end
+
+    it 'has an order total of zero' do
+      expect(order.total).to be_zero
+    end
+
+    it 'has not been placed' do
+      expect(order).not_to be_placed
+    end
+  end
+
+  describe '#read_menu' do
+    it 'can read the menu' do
+      menu = double(:menu, name: "Dominos", dishes: { "Cheese & Tomato" => 6 })
```

Comment:
Yes, but I needed a smaller menu for checking STDOUT more easily. After implementing your previous advice of having a simpler menu, this one will not be needed.